rust/parquet/src/arrow/levels.rs

```diff
@@ -0,0 +1,692 @@
+        let array_mask = vec![true, true];
+
+        // calculate level1 levels
+        let levels = parent_levels.calculate_child_levels(
+            array_offsets.clone(),
+            array_mask.clone(),
+            true,
+            false,
+            1,
+        );
+        //
+        let expected_levels = LevelInfo {
+            definition: vec![1, 1, 1, 1],
+            repetition: Some(vec![0, 1, 0, 1]),
+            definition_mask: vec![(true, 1), (true, 1), (true, 1), (true, 1)],
+            array_offsets,
+            array_mask,
+            max_definition: 1,
+            is_list: true,
+            is_nullable: false,
+        };
+        assert_eq!(levels, expected_levels);
+
+        // level2
+        let parent_levels = levels;
+        let array_offsets = vec![0, 3, 7, 8, 10];
+        let array_mask = vec![true, true, true, true];
+        let levels = parent_levels.calculate_child_levels(
+            array_offsets.clone(),
+            array_mask.clone(),
+            true,
+            false,
+            2,
+        );
+        let expected_levels = LevelInfo {
+            definition: vec![2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+            repetition: Some(vec![0, 2, 2, 1, 2, 2, 2, 0, 1, 2]),
+            definition_mask: vec![
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+                (true, 2),
+            ],
+            array_offsets,
+            array_mask,
+            max_definition: 2,
+            is_list: true,
+            is_nullable: false,
+        };
+        assert_eq!(&levels, &expected_levels);
+    }
+
+    #[test]
+    fn test_calculate_one_level_1() {
+        // This test calculates the levels for a non-null primitive array
+        let parent_levels = LevelInfo {
+            definition: vec![1; 10],
+            repetition: None,
+            definition_mask: vec![(true, 1); 10],
+            array_offsets: (0..=10).collect(),
+            array_mask: vec![true; 10],
+            max_definition: 0,
+            is_list: false,
+            is_nullable: false,
+        };
+        let array_offsets: Vec<i64> = (0..=10).collect();
+        let array_mask = vec![true; 10];
+
+        let levels = parent_levels.calculate_child_levels(
+            array_offsets.clone(),
+            array_mask.clone(),
+            false,
+            false,
+            1,
+        );
+        let expected_levels = LevelInfo {
+            definition: vec![1; 10],
+            repetition: None,
+            definition_mask: vec![(true, 1); 10],
+            array_offsets,
+            array_mask,
+            max_definition: 1,
+            is_list: false,
+            is_nullable: false,
+        };
+        assert_eq!(&levels, &expected_levels);
+    }
+
+    #[test]
+    #[ignore]
```

Comment:
why `#ignore`?

Reply:
still failing as it tests lists, I'll add the reason